cdxev/validator/validate.py

```diff
@@ -22,16 +22,25 @@ def validate_sbom(
     file: Path,
     report_format: t.Optional[str],
     report_path: t.Optional[Path],
-    schema_type: str = "default",
-    filename_regex: t.Optional[str] = "",
-    schema_path: str = "",
+    schema_type: t.Optional[str],
+    filename_regex: t.Optional[str],
+    schema_path: t.Optional[Path],
 ) -> int:
     errors = []
-    if input_format == "json":
-        sbom_schema, used_schema_path = open_schema(
-            sbom, file, schema_type, schema_path
+    if (schema_path is not None) == bool(schema_type):
+        raise AssertionError(
+            "Exactly one of schema_path or schema_type must be non-None"
         )
 
+    if input_format == "json":
+        try:
+            spec_version = sbom["specVersion"]
+        except KeyError:
+            logger.info("SBOM doesn't specify CycloneDX version. Assuming version 1.3.")
```

Comment:
And: if a `schema-path` is provided, there is no assumption about the `specVersion`, than it's the version from the provided schema.

Reply:
You're right, good catch. Thanks!
Will push a fix momentarily.